libretroshare/src/serialiser/rstypeserializer.cc

```diff
@@ -255,13 +255,15 @@ template<> bool RsTypeSerializer::deserialize(const uint8_t data[],uint32_t size
 
     r.first = (uint8_t*)rs_malloc(r.second) ;
 
-    ok = ok && (NULL != r.first);
```

Comment:
The idea is that because C++ has lazy evaluation, the second condition is not tested if ok==false. This is the reason it's written like this. Besides, ok is a bool, so && is the appropriate operator, not &

Reply:
Done